contracts/L2ChainInfo.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.20;
+
+/*
+This contract lives on L2 and shares information about the chain.
+It needs to get this information by being called after a fork.
+We made it for the ForkArbitrator to get the result after a fork. 
+Other contracts may also find it useful.
+It must be called after a fork until it's updated.
+Queries against it will revert until the update is done.
+
+TODO: Should we somehow make it updateable even if you miss the fork window?
+*/
+
+import {IBridgeMessageReceiver} from "@RealityETH/zkevm-contracts/contracts/interfaces/IBridgeMessageReceiver.sol";
+
+contract L2ChainInfo is IBridgeMessageReceiver{
+
+    // These should be fixed addresses that never change
+    address public l2bridge; 
+    address public l1globalRouter;
+    uint32 public originNetwork;
+
+    uint256 internal chainId;
+    address internal forkingManager;
+    uint256 internal forkFee;
+    // uint256 internal totalSupply;
+
+    // Questions are stored by isL2->forker->id
+    // The forker is assumed to have created a unique id within itself for the dispute it's forking over
+    mapping(bool=>mapping(address=>mapping(bytes32=>bytes32))) public forkQuestionResults;
+    mapping(bool=>mapping(address=>mapping(bytes32=>uint256))) public questionToChainID;
+
+    constructor(uint32 _originNetwork, address _l2bridge, address _l1globalRouter) {
+	originNetwork = _originNetwork;
+        l2bridge = _l2bridge; 
+        l1globalRouter = _l1globalRouter;
+    }
+
+    modifier isUpToDate {
```

Comment:
To make your contracts smaller, its nice to factor out such generaliseable components. In this case, I have done it already before, https://github.com/RealityETH/subjectivocracy/pull/74

Reply:
In this case `InitializeChain.sol` does something subtly different. It nearly fooled me... I feel like writing some article like "Solidity code reuse considered harmful"

Reply:
what is the difference? the require statements are exactly the same. Its only that they are named differently, right?

Reply:
The `InitializeChain.sol` is setting `chainId` in the constructor. Doing that here will result in none of the initial variables being set, because it's being used to track whether we set them.